### PR TITLE
Made parent and child commands for Hint Broadcast

### PR DESCRIPTION
**Pull request by 6hundred9**
Didn't know what `Add()` was supposed to be but made the rest @Misfiy

### CODE REVIEW

File: AdminTools/Commands/HintBroadcast/HintBroadcast.cs [hunk 1]
```diff
@@ -22,229 +22,34 @@ public class HintBroadcast : ParentCommand, IUsageProvider
...
+            RegisterCommand(new Group());
+            RegisterCommand(new Groups());
+            RegisterCommand(new User());
+            RegisterCommand(new Users());
+            RegisterCommand(new Random());
+            RegisterCommand(new Staff());
+            RegisterCommand(new Clearall());
         }
 
         protected override bool ExecuteParent(ArraySegment<string> arguments, ICommandSender sender, out string response) // TODO: Make it ParentCommand
```
Comment (Misfiy) [inline AdminTools/Commands/HintBroadcast/HintBroadcast.cs:35]: Make this not do anything, just return response

**Review by Misfiy — CHANGES_REQUESTED** [review]

**Comment by 6hundred9** [comment]
@Misfiy I did the changes

File: AdminTools/Commands/HintBroadcast/Clear.cs [hunk 2]
```diff
@@ -1,13 +1,41 @@
...
+        public bool Execute(ArraySegment<string> arguments, ICommandSender sender, out string response)
+        {
+            if (arguments.Count < 1)
+            {
+                response = "Usage: hbc clear (user ids / usernames)";
+                return false;
+            }
+
+            IEnumerable<Player> ply = Player.GetProcessedData(arguments);
+            if (ply == null)
```
Comment (Misfiy) [inline AdminTools/Commands/HintBroadcast/Clear.cs:23]: wont be null, check if empty instead

File: AdminTools/Commands/HintBroadcast/Clear.cs [hunk 3]
```diff
@@ -1,13 +1,41 @@
...
+            {
+                response = $"Player not found: {arguments.At(0)}";
+                return false;
+            }
+
+            foreach (Player player in ply)
+            {
+                player.ShowHint(" ");
```
Comment (Misfiy) [inline AdminTools/Commands/HintBroadcast/Clear.cs:31]: use string.Empty

File: AdminTools/Commands/HintBroadcast/Clear.cs [hunk 4]
```diff
@@ -1,13 +1,41 @@
...
+            }
+            response = "Cleared hints of users";
+            return true;
+        }
+
+        public string Command { get; } = "clear";
+        public string[] Aliases { get; }
```
Comment (Misfiy) [inline AdminTools/Commands/HintBroadcast/Clear.cs:38]: Array.Empty

File: AdminTools/Commands/HintBroadcast/User.cs [hunk 5]
```diff
@@ -5,10 +5,40 @@
...
+                return false;
+            }
+
+            ply.ShowHint(Extensions.FormatArguments(arguments, 2), time);
+            response = $"Hint sent to {ply.Nickname}";
+            return true;
+        }
+
+        public string Command { get; } = "user";
+        public string[] Aliases { get; }
```
Comment (Misfiy) [inline AdminTools/Commands/HintBroadcast/User.cs:41]: Array.Empty

**Review by Misfiy — CHANGES_REQUESTED** [review]

File: AdminTools/Commands/HintBroadcast/User.cs [hunk 6]
```diff
@@ -5,10 +5,43 @@
...
     {
+        public bool Execute(ArraySegment<string> arguments, ICommandSender sender, [UnscopedRef] out string response)
+        {
+            if (arguments.Count < 3)
+            {
+                response = "Usage: hbc user (player ids / names (ie. 1,2,3 nameless,ced777ric)) (time) (message)";
+                return false;
+            }
+
+            IEnumerable<Player> ply = Player.GetProcessedData(new ArraySegment<string>(arguments.At(0).Split(',')));
```
Comment (Misfiy) [inline AdminTools/Commands/HintBroadcast/User.cs:22]: nuh uh

Player.GetProcessedData(arguments)

**Review by Misfiy — CHANGES_REQUESTED** [review]

**Review by ALEXWARELLC — APPROVED** [review]
👍 Looks good.

**Comment by 6hundred9** [comment]
> 👍 Looks good.

lets goooo exiled-gpt said it looks good

**Comment by 6hundred9** [comment]
@Misfiy @ALEXWARELLC sooo is it gonna be pushed to Exiled-Team:dev?

**Pull request merged**